react/features/noise-suppression/actions.ts

```diff
@@ -0,0 +1,44 @@
+import { SET_NOISE_SUPPRESSION_STATE, TOGGLE_NOISE_SUPPRESSION } from './actionTypes';
+import { isNoiseSuppressionActive } from './functions';
+
+/**
+ * Updates the noise suppression active state.
+ *
+ * @param {boolean} active - Is noise suppression active.
+ * @returns {{
+ *      type: SET_NOISE_SUPPRESSION_STATE,
+ *      active: boolean
+ * }}
+ */
+export function setNoiseSuppressionState(active: boolean) : any {
+    return {
+        type: SET_NOISE_SUPPRESSION_STATE,
+        active
+    };
+}
+
+/**
+ * Signals that noise suppression action has been triggered.
+ *
+ * @returns {{
+ *     type: TOGGLE_NOISE_SUPPRESSION,
+ * }}
+ */
+export function toggleNoiseSuppression() : any {
+    return {
+        type: TOGGLE_NOISE_SUPPRESSION
+    };
+}
+
+/**
+ * Turns off noise suppression if it's already enabled
+ *
+ * @returns {Function}
+ */
+ export function turnOffNoiseSuppression() : Function {
```

Comment:
Feels a bit convoluted, no? I'd say TOGGLE_NOISE_SUPPRESSION should not be the action we act on, but a helper that reads the local state and dispatches the opposite. Then we can simplify this by removing this extra action and acting on SET_NOISE_SUPPRESSION_STATE directly.

Do you see a problem with that approach? I might be missing something.

Reply:
I guess, this might stem from my lacking knowledge of redux/middleware/react. So the reason why I have two of them was so that in case of an error other interested parties (middlewares) wouldn't act on SET_NOISE_SUPPRESSION_STATE, whereas if it's just the one, other middleware's wouldn't know if the outcome was successful or failed and still act on it. But I'm assuming there are ways of doing this with a single action? like preventing it from reaching the reducer in case of an error, not sure how though.

Reply:
Let's sync on this.